src/Maestro/Microsoft.DotNet.Maestro.Tasks.Tests/GetBuildManifestMetadataTests.cs

```diff
@@ -0,0 +1,115 @@
+using System;
+using System.Linq;
+using System.Threading;
+using FluentAssertions;
+using NUnit.Framework;
+
+namespace Microsoft.DotNet.Maestro.Tasks.Tests
+{
+    [TestFixture]
+    public class GetBuildManifestMetadataTests
+    {
+        private PushMetadataToBuildAssetRegistry pushMetadata;
+        private string testManifestFolderPath = @".\TestManifests\"; 
+
+        public GetBuildManifestMetadataTests()
+        {
+            pushMetadata = new PushMetadataToBuildAssetRegistry();
+        }
+        [Test]
+        public void EmptyManifestFolderPath()
+        {
+            Action act = () => pushMetadata.GetBuildManifestsMetadata("", new CancellationToken());
+            act.Should().Throw<ArgumentException>().WithMessage("The path is empty. (Parameter 'path')");
+        }
+
+        [Test]
+        public void ParseBasicManifest()
+        {
+            var data = pushMetadata.GetBuildManifestsMetadata(testManifestFolderPath + "OneManifest", new CancellationToken());
```

Comment:
"CancellationToken.None" is a touch clearer than 'new'.  They behave the same, because the default token isn't cancellable, but the "none" just seems clearer to me when I'm reading code.

Reply:
Good to know, thanks!